### ISSUE TITLE
AJAX 403 Forbidden Error for Autocomplete Fields 

### ISSUE THREAD

**Issue by jimafisk** [CityOfBoston/boston.gov#672]
Temporarily introduced/resolved by saving node as a draft and re-opening draft to edit. 
- Can reproduce as admin
- Can reproduce locally

![ajax](https://cloud.githubusercontent.com/assets/5913244/26116133/1508925c-3a30-11e7-8c53-70960b0378d4.png)

![ajax1](https://cloud.githubusercontent.com/assets/5913244/26116135/1696d6ec-3a30-11e7-8022-1c4187473ecf.png)

**Comment by jimafisk** [comment]
For testing @jduffy2001 created a node with all the problem components: https://edit-stg.boston.gov/node/25556/draft

**Comment by jimafisk** [comment]
Related Issue: https://www.drupal.org/node/2322061

**Comment by jimafisk** [comment]
We should be able to solve this by setting paragraph level permissions. I've started working through this but would love to get additional input @jduffy2001 @cob-hondo 

We need to set the following paragraph (component) level permissions:
- view
- create
- update
- delete

There is no distinction between own/any for these permissions.

These are our roles currently: 
- anonymous user (default Drupal role, this is everyone who visits the site)
- authenticated user (default Drupal role, this is anyone who logs into the site)
- administrator (default Drupal role, this kind of user can do anything on the site without restriction)
- Site Administrator
- Content Editor
- Content Author
- Events Editor
- Press Release Editor
- Status Alert Editor
- Developer
- Guide Author
- Landing Author

So I think we can safely add `view` permission to `anonymous user`. 

In looking at the other roles, it appears they are set up to be limited to certain content types. Each of these content types has a paragraphs reference to our components. Since our components are currently tied only to the node they are created on, there doesn't appear to be risk assigning `create`/`update`/`delete` permissions across the board since these users would still be limited to their content type and adhere to the any/own permissions already assigned to their role for that content type. It also doesn't make sense to assign these paragraph permissions individually by role since all roles need full access to manage their component data. Finally, the components available to editors per content type are still set by the allowed value list of the paragraphs reference field.

**tl;dr** I plan to give `create`/`update`/`delete` permissions to the `authenticated user` role for all of our paragraphs (components). Do you see glaring issues with doing this? Thanks!

**Comment by jduffy2001** [comment]
@jimafisk no, I don't see an issue.

To your point, our permissions are locked down at the content level, not the component level, so this shouldn't be a problem. I can't think of any example of this being an issue.

**Comment by jimafisk** [comment]
@BrutiAnteneh here are some testing steps, but I'm sure @jduffy2001 has additional checks. It'd be great if you two could loop in on this to ensure it's working properly.

Steps:

1. Create a new Article page (/node/add/article)
2. Fill out the required information (Title, body)
3. Click the "Components" tab in the left sidebar
4. Under "Component Type" choose "News & Announcements"
5. Start filling out the autocomplete field (Featured Item)

If using the autocomplete field in the last step returns results to pick from and you do not get a popup window with an error message, then this issue is passing.

**Comment by BrutiAnteneh** [comment]
The issue passed as expected. 

Following @jduffy2001 's recommendation, I tested the issue with multiple Featured Items, saving the page before going back in to choose a new Featured Item.
